src/library/Cardano/Fee.hs

```diff
@@ -128,21 +126,47 @@ newtype FeeEstimator i o = FeeEstimator
 data FeeOptions i o = FeeOptions
     { feeEstimator
         :: FeeEstimator i o
+        -- ^ Estimate fees based on selected inputs and requested outputs.
+
     , dustThreshold
         :: DustThreshold
+        -- ^ Change addresses below the given threshold will be evicted
+        -- from the created transaction. Setting 'dustThreshold' to 0
+        -- removes output equal to 0
+
+    , onDanglingChange
+        :: OnDanglingChange
+        -- ^ What do to when we encouter a dangling change output.
+        -- See 'OnDanglingChange'
     } deriving Generic
 
+-- | We call 'dangling' a change output that would be too expensive to add. This
+-- can happen when a change output is small, but adding it generate more fees
+-- than not having it.
+--
+-- In case where nodes accept slightly unbalanced transactions, we may choose to
+-- save money and keep the transaction slightly unbalanced. In case where node
+-- demands exactly balanced transaction, we have no choice but to add the extra
+-- dangling change output and pay for the extra cost induced.
+data OnDanglingChange
+    = PayAndBalance
+    | SaveMoney
```

Comment:
Perhaps (and this is just a suggestion) this would be easier to explain to users in the form of a **policy** regarding the _balance_:

```suggestion
data BalancePolicy
    = RequirePerfectBalance
        - ^ Generate selections that are perfectly balanced, with the
            trade-off of allowing slightly higher fees.
    | RequireMinimalFee
        - ^ Generate selections with the lowest fees possible, with the
            trade-off of allowing slightly imbalanced selections.
```